### DISCUSSION TITLE
Feature Request: include "context" in the PrimeReact Menu component

### DISCUSSION THREAD

**Discussion by peconomou929** [primefaces/community#1728] [PrimeReact] [answered]
All PrimeReact components have the so-called "Pass Through Method Options" which include a subset of the following three objects: `props`, `state`, and `context`. 

Let's take a look at the `TabMenu` for example. It has all three objects. The `context`, which is of type `TabMenuContext`, has a property `item` and a property `index`. This allows you to query each tab item individually in the pass through options, and as a result, give each tab its own styles. For example, we can make the selected tab look different from the rest. 

This is not the case with the PrimeReact Menu. There is no `context` in the Pass Through Method Options. Hence, there is no way to apply different styles to the different menu items (without creating your own custom JavaScript solution which targets certain elements at run time). This would be useful if I wanted to show a particular menu item as being "selected" (much like a selected tab in a TabMenu). 

Could we therefore add a `context` to the PrimeReact Menu component?

**Comment by melloware** [comment, marked as answer]
Ticket: https://github.com/primefaces/primereact/issues/6489
